### PR TITLE
settlement.txCostUsd

### PR DESCRIPTION
**Pull request by GabrielCamba**
Added missing prop txCostUsd to settlement

@anxolin  @alfetopito Does make sense to add txCostEth to the settlement? we need to do a conversion on gc from xDai to Eth.

### CODE REVIEW

**Comment by alfetopito** [comment]
> Added missing prop txCostUsd to settlement
> 
> @anxolin @alfetopito Does make sense to add txCostEth to the settlement? we need to do a conversion on gc from xDai to Eth.

As we discussed in the meeting, let's add the native and USD prices: txCostNative, txCostUsd

**Review by alfetopito — APPROVED** [review]

File: src/modules/settlements.ts [hunk 1]
```diff
@@ -14,28 +14,27 @@ export namespace settlements {
...
-        // commented code is a WIP will be addressed with:
-        // https://github.com/cowprotocol/subgraph/issues/18
         let DEFAULT_DECIMALS =  BigInt.fromI32(18)
-        let gasPriceUsd = ZERO_BD
+        let txCostUsd = ZERO_BD
+        let txCostNative = ZERO_BD
         if (network == 'xdai') {
-            // txGasPrice is in xdai
-            gasPriceUsd = convertTokenToDecimal(txGasPrice, DEFAULT_DECIMALS)
+            txCostUsd = convertTokenToDecimal(txGasPrice, DEFAULT_DECIMALS)
```
Comment (anxolin) [inline src/modules/settlements.ts:21]: `convertTokenToDecimal(txGasPrice, DEFAULT_DECIMALS)` is done both in the IF/ELSE should u refactor?

File: src/modules/settlements.ts [hunk 2]
```diff
@@ -14,28 +14,27 @@ export namespace settlements {
...
-            //txGasPrice is in eth
-            //let ethPrice = ZERO_BD
-            //ethPrice = getEthPriceInUSD()
-            //let txGasPriceEth = convertTokenToDecimal(txGasPrice, DEFAULT_DECIMALS)
-            //gasPriceUsd = txGasPriceEth.times(ethPrice)
+            // txgasPrice in Eth networks is expressed in eth so we need to do a conversion
+            let ethPrice = getEthPriceInUSD()
+            let txGasPriceEth = convertTokenToDecimal(txGasPrice, DEFAULT_DECIMALS)
+            txCostUsd = txGasPriceEth.times(ethPrice)
+            txCostNative = txGasPriceEth
```
Comment (anxolin) [inline src/modules/settlements.ts:28]: also here would be simplified with the refacrtor

File: schema.graphql [hunk 3]
```diff
@@ -42,6 +42,10 @@ type Token @entity {
   priceUsd: BigDecimal
   "History of trading for this token"
   history: [TokenTradingEvent!]! @derivedFrom(field: "token")
+  "Hourly totals"
+  hourlyTotals: [TokenDailyTotal!]! @derivedFrom(field: "token")
+  "Daily totals"
+  dailyTotals: [TokenDailyTotal!]! @derivedFrom(field: "token")
```
Comment (anxolin) [inline schema.graphql:48]: Is this an unrelared change?

Comment (alfetopito) [reply schema.graphql:48]: Also, both hourly and daily values are assigned TokenDailyTotal

File: src/modules/settlements.ts (hunk 1 above)
Comment (GabrielCamba) [reply src/modules/settlements.ts]: Great! solved!

File: schema.graphql (hunk 3 above)
Comment (GabrielCamba) [reply schema.graphql:48]: My bad! already solved in other commit and merged branch

**Pull request merged**